Zend/tests/pipe_operator/call_by_ref.phpt

```diff
@@ -0,0 +1,20 @@
+--TEST--
+Pipe operator accepts by-reference functions
+--FILE--
+<?php
+
+function _modify(int &$a): string {
+    $a += 1;
+    return "foo";
+}
+
+//try {
+$a = 5;
+$res1 = $a |> '_modify';
```

Comment:
I think here is so much visual separation that I would actually expect a deref in between. I cannot really imagine why you would expect pipe to pass references. I consider pipe more like "pass value to function and call it" than "pass variable to function and call it". Should be noted in the RFC obviously then.

Reply:
Yeah, I think I would also prefer references being an error in this context. I was going off the wording in the RFC here:

>  Functions that accept their first parameter by reference are supported, and will behave exactly as if they were called in the normal “inside out” fashion. However, unless they return a value as well they are not of much use.

Reply:
I'm perfectly OK blocking by-ref variables here, if I can figure out how to do so.  It was more a matter of "the implementation allows it, so I should document that."  I cannot think of a use case where you'd actually want to use a reference in pipes.

Reply:
Actually, correction.  I talked with @krakjoe, and it sounds like blocking references is going to require a completely different implementation with new opcodes.  So, I'm not going to do that, because it's more work than I think is justified.

I'll see about adding more tests to see what happens with references, but I think just building on existing machinery and KISS is the best approach.